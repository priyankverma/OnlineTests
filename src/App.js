import "./App.css";
import React, { Component } from "react";
import { createStore, applyMiddleware, compose } from "redux";
import { Provider } from "react-redux";
import combinedReducer from "./../src/redux/combinedReducers";

import {
  BrowserRouter as Router,
  Switch,
  Route,
  withRouter,
  Redirect,
} from "react-router-dom";
import Dashboard from "../src/screens/dashboard/dashboard";
import LoginScreen from "./screens/login/loginScreen";
import Summary from "./screens/summary/summary";
import PrivateRoute from "./commonComponents/privateRoute/privateRoute";
const ReduxThunk = require("redux-thunk").default;

let composeEnhancer = compose;
export const store = createStore(
  //creates an instance of store to be used globally
  combinedReducer, // pass in the combined reducers
  composeEnhancer(applyMiddleware(ReduxThunk)) // use the middleware for async actions
);

class App extends Component {
  constructor(props) {
    super(props);

    // Store the previous pathname and search strings
    this.currentPathname = null;
    this.currentSearch = null;
  }
  componentDidMount() {
    const { history } = this.props;
    history.listen((newLocation, action) => {
      if (action === "REPLACE") {
        if (
          newLocation.pathname !== this.currentPathname ||
          newLocation.search !== this.currentSearch
        ) {
          // Save new location
          this.currentPathname = newLocation.pathname;
          this.currentSearch = newLocation.search;

          // Clone location object and push it to history
          history.replace({
            pathname: newLocation.pathname,
            search: newLocation.search,
          });
        }
      } else {
        // Send user back if they try to navigate back
        history.go(1);
      }
    });
  }

  render() {
    console.log("signupReducer", this.props.signupReducer);
    return (
      <Provider store={store}>
        <Router>
          <Switch>
            <Route
              exact
              path="/"
              render={() => <Redirect to="/loginScreen" />}
            />

            <Route exact path="/loginScreen" component={LoginScreen} />
            <PrivateRoute path="/dashboard">
              <Dashboard />
            </PrivateRoute>
            <PrivateRoute path="/summary">
              <Summary />
            </PrivateRoute>
            <Route path="/dashboard" component={Dashboard} />
            <Route path="/summary" component={Summary} />
          </Switch>
        </Router>
      </Provider>
    );
  }
}

// export default connect(mapStateToProps, null)(App);
export default withRouter(App);
