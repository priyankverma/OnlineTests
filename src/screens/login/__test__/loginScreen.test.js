import { fireEvent, render, screen } from "@testing-library/react";
import * as React from "react";
import LoginScreen from "./../loginScreen";
import { Provider } from "react-redux";
import { store } from "./../../../App";
import renderer from "react-test-renderer";

describe("Login Component", () => {
  let LoginComponent = (
    <Provider store={store}>
      <LoginScreen />
    </Provider>
  );
  // render(LoginComponent);
  test("username textInput must set username correctly", () => {
    render(LoginComponent);
    const usernameInput = screen.getByPlaceholderText("eg: john.doe");
    expect(usernameInput).toHaveValue(""); // checks for initial value to be ""

    fireEvent.change(usernameInput, { target: { value: "testName" } });
    expect(usernameInput).toHaveValue("testName"); // checks by passing a value testName
  });

  test("password textInput must set password correctly", () => {
    render(LoginComponent);
    const passwordInput = screen.getByPlaceholderText("Password");
    expect(passwordInput).toHaveValue(""); // checks for initial value to be ""

    fireEvent.change(passwordInput, { target: { value: "testPassword" } });
    expect(passwordInput).toHaveValue("testPassword"); // checks by passing a value testPassword
  });

  it("matches snapshot perfectly for loginscreen", () => {
    const tree = renderer.create(LoginComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
