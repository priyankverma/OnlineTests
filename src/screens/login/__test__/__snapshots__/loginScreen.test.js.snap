// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component matches snapshot perfectly for loginscreen 1`] = `
<div
  className="sc-bdvvaa BhqoN"
  height={768}
>
  <div
    className="sc-gsDJrp egHnjR"
  >
    <div
      className="sc-hKwCoD faqGRQ"
    >
      <div
        className="sc-dkPtyc jVJYKn"
      >
        <h1>
          Registration
        </h1>
        <img
          alt={Object {}}
          src="substract.png"
        />
        <h4>
          You need to register in order to procee
        </h4>
        <div
          className="sc-jrQzUz ltEOZ"
        >
          <h4>
            Your name here
          </h4>
          <input
            className="sc-kDThTU gVVStl"
            height="30px"
            name="name"
            onChange={[Function]}
            placeholder="Name"
            type="text"
          />
        </div>
        <div
          className="sc-jrQzUz ltEOZ"
        >
          <h4>
            Password
          </h4>
          <input
            className="sc-kDThTU ibWfVL"
            height="30px"
            name="name"
            onChange={[Function]}
            placeholder="Password"
            type="password"
          />
          <label
            className="ant-checkbox-wrapper"
            style={
              Object {
                "fontSize": 13,
                "fontWeight": 500,
              }
            }
          >
            <span
              className="ant-checkbox"
              style={Object {}}
            >
              <input
                checked={false}
                className="ant-checkbox-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                type="checkbox"
              />
              <span
                className="ant-checkbox-inner"
              />
            </span>
            <span>
               
              Show Password
            </span>
          </label>
        </div>
        <div
          className="sc-iqsfdx dJzaLt"
          data-testid="button"
          height="15px"
          onClick={[Function]}
          width="40%"
        >
          <b>
            Proceed
          </b>
        </div>
        <div
          className="sc-crHlIS kLhkLr"
        >
          <b
            className="sc-egiSv kjROWm"
          >
            <hr />
            <b>
              FOLLOW US ON
            </b>
            <hr />
          </b>
          <div
            className="sc-bqiQRQ SZbYN"
          >
            <div
              className="sc-ksdxAp QUrnf"
              onClick={[Function]}
            >
              <b
                className="sc-hBURRC fqFsmD"
              >
                <img
                  alt={Object {}}
                  src="twitter.png"
                />
                Twitter
              </b>
            </div>
            <div
              className="sc-ksdxAp QUrnf"
              onClick={[Function]}
            >
              <b
                className="sc-hBURRC dvfuTS"
              >
                <img
                  alt={Object {}}
                  src="facebook.png"
                />
                Facebook
              </b>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-eCImvq dbOFKR"
    >
      <img
        className="sc-jRQAMF fPNhRG"
        src="exams.png"
      />
    </div>
  </div>
</div>
`;
